src/sinks/sematext/logs.rs

```diff
@@ -106,30 +106,35 @@ impl SinkConfig for SematextLogsConfig {
 }
 
 struct MapTimestampStream {
-    inner: Box<dyn StreamSink + Send>,
+    inner: Box<dyn StreamSink<EventArray> + Send>,
 }
 
 #[async_trait]
-impl StreamSink for MapTimestampStream {
-    async fn run(self: Box<Self>, input: BoxStream<'_, Event>) -> Result<(), ()> {
+impl StreamSink<EventArray> for MapTimestampStream {
+    async fn run(self: Box<Self>, input: BoxStream<'_, EventArray>) -> Result<(), ()> {
         let mapped_input = input.map(map_timestamp).boxed();
         self.inner.run(mapped_input).await
     }
 }
 
 /// Used to map `timestamp` to `@timestamp`.
-fn map_timestamp(mut event: Event) -> Event {
-    let log = event.as_mut_log();
-
-    if let Some(ts) = log.remove(crate::config::log_schema().timestamp_key()) {
-        log.insert("@timestamp", ts);
-    }
-
-    if let Some(host) = log.remove(crate::config::log_schema().host_key()) {
-        log.insert("os.host", host);
+fn map_timestamp(mut events: EventArray) -> EventArray {
+    match &mut events {
+        EventArray::Logs(logs) => {
+            for log in logs {
+                if let Some(ts) = log.remove(crate::config::log_schema().timestamp_key()) {
+                    log.insert("@timestamp", ts);
+                }
+
+                if let Some(host) = log.remove(crate::config::log_schema().host_key()) {
+                    log.insert("os.host", host);
+                }
+            }
+        }
+        _ => unreachable!("This sink only accepts logs"),
```

Comment:
I'm not quite sure I understand: why does this take `EventArray` instead of `Event`?

Reply:
The `sematext_logs` sink is just a wrapper for `elasticsearch` which, after conversion, is now handling `EventArray` items in its stream. So, this `map` function has to also handle that. One small place where that internal detail has to leak out.

Reply:
Gotcha. 👍🏻